packages/typescript-estree/src/ast-converter.ts

```diff
@@ -16,7 +16,7 @@ export function astConverter(
    * source.
    */
   // internal typescript api...
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment
   const parseDiagnostics = (ast as any).parseDiagnostics;
```

Comment:
I think we could just add `parseDiagnostics` to our internal types here
https://github.com/typescript-eslint/typescript-eslint/blob/20f05f659d5cb219cfc7f7212679c0a17811b3e7/packages/typescript-estree/typings/typescript.d.ts#L3-L8